core/notifications/notifications.go

```diff
@@ -1,45 +1,135 @@
+	"github.com/anyproto/anytype-heart/core/domain"
+	sb "github.com/anyproto/anytype-heart/pkg/lib/core/smartblock"
+	"github.com/anyproto/anytype-heart/pkg/lib/logging"
+	"github.com/anyproto/anytype-heart/space"
+
+	"github.com/anyproto/anytype-heart/core/block/editor/smartblock"
+	"github.com/anyproto/anytype-heart/core/block/editor/state"
 	"github.com/anyproto/anytype-heart/core/event"
 	"github.com/anyproto/anytype-heart/pb"
 	"github.com/anyproto/anytype-heart/pkg/lib/pb/model"
 )
 
+var log = logging.Logger("notifications")
+
 const CName = "notificationService"
 
 type Notifications interface {
-	app.Component
-	CreateAndSendLocal(notification *model.Notification) error
+	app.ComponentRunnable
+	CreateAndSend(notification *model.Notification) error
 	UpdateAndSend(notification *model.Notification) error
 	Reply(notificationID []string, notificationAction model.NotificationActionType) error
 	List(limit int64, includeRead bool) ([]*model.Notification, error)
 }
 
 type notificationService struct {
-	eventSender       event.Sender
-	notificationStore NotificationStore
+	notificationID     string
+	notificationCh     chan struct{}
+	notificationErr    error
+	notificationCancel context.CancelFunc
+	eventSender        event.Sender
+	notificationStore  NotificationStore
+	spaceService       space.Service
+	picker             block.ObjectGetter
 }
 
 func New() Notifications {
-	return &notificationService{}
+	return &notificationService{
+		notificationCh: make(chan struct{}),
+	}
 }
 
 func (n *notificationService) Init(a *app.App) (err error) {
 	n.notificationStore = app.MustComponent[NotificationStore](a)
 	n.eventSender = app.MustComponent[event.Sender](a)
+	n.spaceService = app.MustComponent[space.Service](a)
+	n.picker = app.MustComponent[block.ObjectGetter](a)
 	return nil
 }
 
 func (n *notificationService) Name() (name string) {
 	return CName
 }
 
-func (n *notificationService) CreateAndSendLocal(notification *model.Notification) error {
+func (n *notificationService) Run(_ context.Context) (err error) {
+	notificationContext, notificationCancel := context.WithCancel(context.Background())
+	n.notificationCancel = notificationCancel
+	go n.loadNotificationObject(notificationContext)
+	go n.indexNotifications(notificationContext)
+	return nil
+}
+
+func (n *notificationService) indexNotifications(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			log.Errorf("failed to index notifications: %v", ctx.Err())
+			return
+		case <-n.notificationCh:
+			if n.notificationErr != nil {
+				log.Errorf("failed to get notification object: %v", n.notificationErr)
+				return
+			}
+			n.updateNotificationsInLocalStore()
+			return
+		}
+	}
+}
+
+func (n *notificationService) updateNotificationsInLocalStore() {
+	var notifications map[string]*model.Notification
+	err := block.DoState(n.picker, n.notificationID, func(s *state.State, sb smartblock.SmartBlock) error {
```

Comment:
Maybe it is enough to call `block.Do` generic here? 
State could be retrived from sb.NewState()
`DoState` invokes sb.Apply(s), while `Do` does not

Reply:
We need to call apply to push changes of notifications

Reply:
My bad, thank you - fixed it